### PR TITLE
feat: Agent REST: Support to configure databases

### LINKED ISSUE #2156: Agent REST: Support to configure databases
Currently, aries-rest only supports level db. The support for mysql and couchdb is available in sdk. Needs to extend this to aries-rest as well.

### PR DESCRIPTION
**Pull request by fqutishat**
closes #2156

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/storage/couchdb/couchdbstore.go [hunk 1]
```diff
@@ -64,6 +64,11 @@ func NewProvider(hostURL string, opts ...Option) (*Provider, error) {
 
 	p := &Provider{hostURL: hostURL, couchDBClient: client, dbs: map[string]*CouchDBStore{}}
 
+	_, err = client.Ping(context.Background())
```
Comment (llorllale) [inline pkg/storage/couchdb/couchdbstore.go:67]: please fix the unit tests as well... there's at least one where no error is expected after providing a `wrongURL`.

Comment (fqutishat) [reply pkg/storage/couchdb/couchdbstore.go:67]: done

File: cmd/aries-agent-rest/startcmd/start.go [hunk 2]
```diff
@@ -46,12 +52,25 @@ const (
 	agentTokenFlagUsage     = "Check for bearer token in the authorization header (optional)." +
 		" Alternatively, this can be set with the following environment variable: " + agentTokenEnvKey
 
-	// db path flag
-	agentDBPathFlagName      = "db-path"
-	agentDBPathEnvKey        = "ARIESD_DB_PATH"
-	agentDBPathFlagShorthand = "d"
-	agentDBPathFlagUsage     = "Path to database." +
-		" Alternatively, this can be set with the following environment variable: " + agentDBPathEnvKey
+	databaseTypeFlagName      = "database-type"
```
Comment (llorllale) [inline cmd/aries-agent-rest/startcmd/start.go:55]: There are many parameters already. Suggest we merge this one into databaseURL in a `type://url` format... see https://github.com/trustbloc/edge-sandbox/blob/4b1f867848ced0394ba37172b919ef4898cebec8/cmd/common/common.go#L40-L46

Comment (fqutishat) [reply cmd/aries-agent-rest/startcmd/start.go:55]: it's better to separate less complexity take look at   https://github.com/trustbloc/edge-service/blob/master/cmd/vc-rest/startcmd/start.go#L96

File: cmd/aries-agent-rest/startcmd/start.go [hunk 3]
```diff
@@ -46,12 +52,25 @@ const (
...
+	databaseTypeEnvKey        = "ARIESD_DATABASE_TYPE"
+	databaseTypeFlagShorthand = "q"
+	databaseTypeFlagUsage     = "The type of database to use for everything except key storage. " +
+		"Supported options: mem, couchdb, mysql. " +
+		" Alternatively, this can be set with the following environment variable: " + databaseTypeEnvKey
```
Comment (rolsonquadras) [inline cmd/aries-agent-rest/startcmd/start.go:60]: We have `leveldb` as well in the sdk. Don't we need to support that as well ?

Comment (fqutishat) [reply cmd/aries-agent-rest/startcmd/start.go:60]: I will add it in next PR

**Review by rolsonquadras — APPROVED** [review]

File: cmd/aries-agent-rest/startcmd/start.go (hunk 3 above)
Comment (fqutishat) [reply cmd/aries-agent-rest/startcmd/start.go:62]: I add it

**Review by sudeshrshetty — APPROVED** [review]

**Review by llorllale — APPROVED** [review]

**Pull request merged**